include/object/reply.h

```diff
@@ -13,12 +13,16 @@
 /* Unlink a reply from its tcb */
 static inline void reply_unlink(reply_t *reply, tcb_t *tcb)
 {
+    /* check that the tcb has a thread state with reply */
+    assert(thread_state_get_tsType(tcb->tcbState) == ThreadState_BlockedOnReceive ||
+           thread_state_get_tsType(tcb->tcbState) == ThreadState_BlockedOnReply);
+
     /* check the tcb and reply are linked correctly */
     assert(reply->replyTCB == tcb);
     assert(thread_state_get_replyObject(tcb->tcbState) == REPLY_REF(reply));
 
-    thread_state_ptr_set_replyObject(&tcb->tcbState, REPLY_REF(0));
     reply->replyTCB = NULL;
+    /* This means the value of the thread state reply reference no longer matters. */
```

Comment:
It goes against my sensibilities not to clear the pointer, even if no one is supposed to look at it. :-(